### DISCUSSION TITLE
How to communicate with a can bus over a usb serial port?

### DISCUSSION THREAD

**Discussion by AngelHadzhiev** [hardbyte/python-can#1722] [Q&A]
Hi everybody, 

I have found this tool recently. I need to read CAN bus messages that are passed through a USB serial port on a Linux computer. In the documentation there is not much I could find regarding communication over USB serial port. 

I have this at the moment: 

```
with can.interface.Bus(interface="serial", 
                                     channel="/dev/ttyACM0") 
                                     as bus:                         
       message = can.Message(arbitration_id=0x{some id}), data=["some data"])
       
       try: 
            bus.send(message)
            
           for msg in bus: 
                 print(msg.data)
                 
                 
       except Exception as e: 
            print(e)
```
       
       
 My code does not break but also nothing happens. I am sure this configuration is correct because with Putty it works fine. Can someone help me out or refer me to some documentation. Thanks in advance.

**Comment by grant-allan-ctct** [comment]
In your question, you have `channel-"/dev/ttyACM0"` and I wonder whether it is like that in your code also, because the `-` character should be `=` shouldn't it?

**Reply by AngelHadzhiev** [reply]
Hi, it is an "=" I have misspelled it here. 

I can clarify that when I use 'pyserial' and not python-can, I can read the messages coming on the port but I am unable to send or at least I don't know how. When I do it with the bus example here in the documentation, I can neither read nor write.
